### DISCUSSION TITLE
Use plain scss -> bundle it to css -> reference css in components?

### DISCUSSION THREAD

**Discussion by gregg-cbs** [vercel/next.js#35186] [Help]
### Heres my idea...

Seeing as components can only import css files and not sass files I want to edit the webpack config to have similar functionality to [nuxt ](https://nuxtjs.org/docs/configuration-glossary/configuration-build/#extractcss)where it exports scss files to separate css files.

I would import a sass or many sass files into `_app.js` or tell webpack where to find my sass files and webpack would export those sass files into separate css files or it would know any scss files without underscores should be its own file.

Then i would like to reference that css file in my components so i have a normal sass implementation. This way I can control what gets bundled into the main scss file and what gets split into its own file - i have control of my css splitting and I get to use plain sass.

### Why not use css modules or css in js?

1. CSS modules is not all that, i have used it extensively and it has its bugbears, so i switched to css in js.
2. Styled Components or CSS-in-JS might decrease css bundle size but it increases page load time because js has to load styles into the head and js is blocking causing some of my page load times to be longer.
3. In a world where the difference of 1s on your load time means making the cut or not and developer experience matters, its come down to this for me. I cannot risk the css in js penalties and the nuances of css modules. I am seeking for a more standard scss implementation.

Please comment here if you have any ideas! Thanks

**Comment by gregg-cbs** [comment]
Alternatively what i might do is write plain css and use css variables. At least gives me reusable variables. Wont have mixins and all the rest though :(

**Comment by icyJoseph** [comment]
Worth nothing that there's support for `module.scss` and `module.sass` to be imported at component level, https://nextjs.org/docs/basic-features/built-in-css-support#sass-support. 

I once worked on a project which used Bulma, which lived on an independent toolchain. It was not ideal, but as a team we rarely ever changed those source files, so we had it living in our project, and built it as a good old SASS project, and the output of it was also part of the git history, and in turn the output was used in the Next App (this was Next v8).

**Reply by gregg-cbs** [reply]
I mentioned in my original post that I know of modules and css in js... problem is neither of those are ideal solutions.

To further add to this, iv today inherited a project (as I am a contractor) and they have done this with the CSS:
![image](https://user-images.githubusercontent.com/57114335/157865966-d1d2ab41-12dc-4a5b-8bb9-c9cf5f9b218a.png)

And included it like this:
![image](https://user-images.githubusercontent.com/57114335/157866024-735737e5-4fd2-48a7-be66-42774706e315.png)

This project is dynamic and will end up having hundreds of pages, at this rate the css is going to blow up like crazy.
I cant go refactor this now.

This would be manageable if I could bundle the css per page, main bundle etc so I could split it up and dynamically load page css if the page is visited.

Not a crazy ask seeing as thats how normal css should work, but i understand thats not how next works. Would be nice if normal sass / css implementation could work.

**Reply by icyJoseph** [reply]
Aha, I think I mixed up components/pages.

Just to make sure we are on the same page, if the stylesheets were externally hosted, you'd use Head+`link` combo on each page, and inject the stylesheets at page level, right?

One thing I might do, is to build the per page SASS to CSS and place it on `public`, and from each page put a `link` that points to them. 

But yeah, now I see your problem, I'll a do a bit of reading to see if there's any other options.
